sources/api/static-pods/src/static_pods.rs

```diff
@@ -0,0 +1,320 @@
+    if path.exists() {
+        fs::remove_file(path).context(error::ManifestDelete { name })?;
+    }
+
+    Ok(())
+}
+
+/// Store the args we receive on the command line
+struct Args {
+    log_level: LevelFilter,
+    socket_path: PathBuf,
+}
+
+/// Print a usage message in the event a bad arg is passed
+fn usage() -> ! {
+    let program_name = env::args().next().unwrap_or_else(|| "program".to_string());
+    eprintln!(
+        r"Usage: {}
+            [ --socket-path PATH ]
+            [ --log-level trace|debug|info|warn|error ]
+
+    Socket path defaults to {}",
+        program_name, DEFAULT_API_SOCKET,
+    );
+    process::exit(2);
+}
+
+/// Prints a more specific message before exiting through usage().
+fn usage_msg<S: AsRef<str>>(msg: S) -> ! {
+    eprintln!("{}\n", msg.as_ref());
+    usage();
+}
+
+/// Parse the args to the program and return an Args struct
+fn parse_args(args: env::Args) -> Args {
+    let mut log_level = None;
+    let mut socket_path = None;
+
+    let mut iter = args.skip(1);
+    while let Some(arg) = iter.next() {
+        match arg.as_ref() {
+            "--log-level" => {
+                let log_level_str = iter
+                    .next()
+                    .unwrap_or_else(|| usage_msg("Did not give argument to --log-level"));
+                log_level = Some(LevelFilter::from_str(&log_level_str).unwrap_or_else(|_| {
+                    usage_msg(format!("Invalid log level '{}'", log_level_str))
+                }));
+            }
+
+            "--socket-path" => {
+                socket_path = Some(
+                    iter.next()
+                        .unwrap_or_else(|| usage_msg("Did not give argument to --socket-path"))
+                        .into(),
+                )
+            }
+
+            _ => usage(),
+        }
+    }
+
+    Args {
+        log_level: log_level.unwrap_or_else(|| LevelFilter::Info),
+        socket_path: socket_path.unwrap_or_else(|| DEFAULT_API_SOCKET.into()),
+    }
+}
+
+fn handle_static_pod<S>(name: S, pod_info: &model::StaticPod) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    // Get basic settings, as retrieved from API.
+    let name = name.as_ref();
+    let enabled = pod_info.enabled.context(error::MissingField {
+        name,
+        field: "enabled",
+    })?;
+    let manifest = pod_info.manifest.as_ref().context(error::MissingField {
+        name,
+        field: "manifest",
+    })?;
+
+    if enabled {
+        let manifest = base64::decode(manifest.as_bytes()).context(error::Base64Decode {
+            base64_string: manifest.as_ref(),
+        })?;
+
+        info!(
+            "Handling static pod '{}' which is enabled: {}",
+            name, enabled
+        );
+
+        // Write the manifest file for this static pod
+        write_manifest_file(name, manifest)?;
+    } else {
+        // Delete the manifest file so the static pod no longer runs (disabled)
+        delete_manifest_file(name)?;
```

Comment:
Does kubelet monitor these files?  If you delete the file, is kubelet smart enough to stop the pod?  Same for file updates; is kubelet smart enought to stop/start the pod with new configuration, etc.?

Reply:
Yes, kubelet monitors the pod manifest path for changes. kubelet does stop the pod if the corresponding manifest is deleted. It also knows when to restart the pod if the manifest changes.